### PR TITLE
Block-Change & Block-Place things logged

### PR DESCRIPTION
**Pull request by Altirix**
- Cauldron - Missing playerentity for some logs as method lacks playerentity. wont be adding those but methods have been found. no clue how im supposed to store the player entity from the interface. should log pretty much everything, from dripstone and weather filling them to players removing with bottles

- Cake - logs eating and placing a candle

- Candle - logs extinguish and ignite from projectiles, flint and steel already logged but was lacking source

- Anvil - On use damage interaction

- JukeBox - Removal and insertion of Discs

- [Grass,Mycelium,Dirt Path,...] -> Dirt trample logging from block placing on top

- Sponge - was logging its own state multiple times and i think was missing some logging on water after placing the sponge. Make nether drying wet sponge place the log the correct block and change it with BlockChange - Actions are not ordered. only certain sponge events get logged in order.

- Farmland - minor change converting to dirt from lack of water is a Dry source rather than Trample

- End Crystal - minor change to include Projectile owners to be logged

- Scaffolding - use the correct blockpos on placement, add an on breaking log when bottom scaffold has been broken, does not have player source only gravity

- Dripstone - Log gravity breaking blocks when dripstone is pointing up

- Campfires - Log interactions.

All still need testing to ensure they fully work, only been tested in dev environment for a small amount of time.

### CODE REVIEW

File: src/main/java/com/github/quiltservertools/ledger/mixin/blocks/cauldron/CauldronBehaviorMixin.java
Comment (yitzy299) [inline src/main/java/com/github/quiltservertools/ledger/mixin/blocks/cauldron/CauldronBehaviorMixin.java:29]: Will need to remove these log calls

File: src/main/kotlin/com/github/quiltservertools/ledger/utility/Sources.kt
Comment (yitzy299) [inline src/main/kotlin/com/github/quiltservertools/ledger/utility/Sources.kt:18]: Just `drip`

File: src/main/kotlin/com/github/quiltservertools/ledger/utility/Sources.kt
Comment (yitzy299) [inline src/main/kotlin/com/github/quiltservertools/ledger/utility/Sources.kt:19]: Source should be `snow`

File: src/main/kotlin/com/github/quiltservertools/ledger/utility/Sources.kt
Comment (yitzy299) [inline src/main/kotlin/com/github/quiltservertools/ledger/utility/Sources.kt:23]: Comes under `fire`

File: src/main/java/com/github/quiltservertools/ledger/mixin/blocks/JukeBoxBlockMixin.java
Comment (yitzy299) [inline src/main/java/com/github/quiltservertools/ledger/mixin/blocks/JukeBoxBlockMixin.java:37]: Jukebox sources should be `Interact`

File: src/main/java/com/github/quiltservertools/ledger/mixin/MusicDiscItemMixin.java
Comment (yitzy299) [inline src/main/java/com/github/quiltservertools/ledger/mixin/MusicDiscItemMixin.java:24]: Is this needed?

File: src/main/java/com/github/quiltservertools/ledger/mixin/BlockItemMixin.java
Comment (yitzy299) [inline src/main/java/com/github/quiltservertools/ledger/mixin/BlockItemMixin.java:45]: Could this be handled with a mixin to sponges?

File: src/main/java/com/github/quiltservertools/ledger/mixin/blocks/FarmlandBlockMixin.java
Comment (yitzy299) [inline src/main/java/com/github/quiltservertools/ledger/mixin/blocks/FarmlandBlockMixin.java:33]: Should be a space between the last two arguments. I'm inclined to stick with `trample` for all farmland to dirt conversions

File: src/main/java/com/github/quiltservertools/ledger/mixin/FlintAndSteelItemMixin.java
Comment (yitzy299) [inline src/main/java/com/github/quiltservertools/ledger/mixin/FlintAndSteelItemMixin.java:29]: Please revert this to `fire`

File: src/main/java/com/github/quiltservertools/ledger/mixin/blocks/CandleBlockMixin.java
Comment (yitzy299) [inline src/main/java/com/github/quiltservertools/ledger/mixin/blocks/CandleBlockMixin.java:43]: `fire`

**Review by yitzy299 — CHANGES_REQUESTED** [review]
A few small changes requested, mostly looks good. Could do with a cleanup of comments and disabled code

File: src/main/java/com/github/quiltservertools/ledger/mixin/BlockMixin.java
Comment (yitzy299) [inline src/main/java/com/github/quiltservertools/ledger/mixin/BlockMixin.java:24]: Will be fixed at some point soon (tm)

File: src/main/java/com/github/quiltservertools/ledger/mixin/CampfireBlockEntityMixin.java
Comment (yitzy299) [inline src/main/java/com/github/quiltservertools/ledger/mixin/CampfireBlockEntityMixin.java:34]: Spaces between arguments here please

File: src/main/java/com/github/quiltservertools/ledger/mixin/EndCrystalEntityMixin.java
Comment (yitzy299) [inline src/main/java/com/github/quiltservertools/ledger/mixin/EndCrystalEntityMixin.java:28]: Please use `source.getSource() instanceof PlayerEntity player` instead of manually casting

File: src/main/java/com/github/quiltservertools/ledger/mixin/EndCrystalEntityMixin.java
Comment (yitzy299) [inline src/main/java/com/github/quiltservertools/ledger/mixin/EndCrystalEntityMixin.java:30]: Same as comment above

File: src/main/java/com/github/quiltservertools/ledger/mixin/MusicDiscItemMixin.java
Comment (yitzy299) [inline src/main/java/com/github/quiltservertools/ledger/mixin/MusicDiscItemMixin.java:35]: `Sources.INTERACT`

File: src/main/java/com/github/quiltservertools/ledger/mixin/blocks/CakeBlockMixin.java
Comment (yitzy299) [inline src/main/java/com/github/quiltservertools/ledger/mixin/blocks/CakeBlockMixin.java:66]: `Sources.INTERACT`

File: src/main/java/com/github/quiltservertools/ledger/mixin/blocks/CampfireBlockMixin.java
Comment (yitzy299) [inline src/main/java/com/github/quiltservertools/ledger/mixin/blocks/CampfireBlockMixin.java:50]: I don't think the comment is needed

File: src/main/java/com/github/quiltservertools/ledger/mixin/blocks/SpongeBlockMixin.java
Comment (yitzy299) [inline src/main/java/com/github/quiltservertools/ledger/mixin/blocks/SpongeBlockMixin.java:54]: Shift is not needed with modifyargs, see some of the other mixins

File: src/main/java/com/github/quiltservertools/ledger/mixin/blocks/SpongeBlockMixin.java
Comment (yitzy299) [inline src/main/java/com/github/quiltservertools/ledger/mixin/blocks/SpongeBlockMixin.java:45]: Same as below

File: src/main/java/com/github/quiltservertools/ledger/mixin/blocks/SpongeBlockMixin.java
Comment (yitzy299) [inline src/main/java/com/github/quiltservertools/ledger/mixin/blocks/SpongeBlockMixin.java:24]: Seem to be a lot of questions here - I hope those can be answered before merging

File: src/main/java/com/github/quiltservertools/ledger/mixin/blocks/cauldron/CauldronBehaviorMixin.java
Comment (yitzy299) [inline src/main/java/com/github/quiltservertools/ledger/mixin/blocks/cauldron/CauldronBehaviorMixin.java:59]: `Sources.DRAIN`

File: src/main/resources/ledger.mixins.json
Comment (yitzy299) [inline src/main/resources/ledger.mixins.json:2]: Please set the IDE formatter to 4 spaces per tab

File: src/main/java/com/github/quiltservertools/ledger/mixin/AnvilScreenHandlerMixin.java
Comment (Altirix) [reply src/main/java/com/github/quiltservertools/ledger/mixin/AnvilScreenHandlerMixin.java:41]: not sure if i can use modifyargs, unless its possible to get `world` and i just never realised it, is capturing local blockstates worse than getting the blockstates back from the world?

File: src/main/resources/ledger.mixins.json
Comment (Altirix) [reply src/main/resources/ledger.mixins.json]: should be fixed, had json defaulting to 2 spaces.

File: src/main/java/com/github/quiltservertools/ledger/mixin/BlockItemMixin.java
Comment (Altirix) [reply src/main/java/com/github/quiltservertools/ledger/mixin/BlockItemMixin.java]: theres some changes ive left in unless you want it fully reverted, mainly just for

`context.getWorld().getBlockEntity(context.getBlockPos()) != null ? context.getWorld().getBlockEntity(context.getBlockPos()) : null,`

and some minor stuff with getting `World`, `pos` and `player` in their own variables

File: src/main/java/com/github/quiltservertools/ledger/mixin/AnvilScreenHandlerMixin.java
Comment (Altirix) [reply src/main/java/com/github/quiltservertools/ledger/mixin/AnvilScreenHandlerMixin.java:38]: oh yeah so i can... just had a look at DaylightDetector

**Review by yitzy299 — APPROVED** [review]
I'm happy with this, will need to wait for @PotatoPresident to review

**Review by PotatoPresident — CHANGES_REQUESTED** [review]
Why was `java.util.Random` imported everywhere

**Comment by Altirix** [comment]
> Why was `java.util.Random` imported everywhere

seems `Random` was always there, however when it went thru the reformatted it moved them down into its own block of imports.

**Review by yitzy299 — APPROVED** [review]

**Review by PotatoPresident — APPROVED** [review]

**Pull request merged**
